java/core/src/main/java/sleeper/core/statestore/FileReferenceStore.java

```diff
@@ -50,6 +50,20 @@ public interface FileReferenceStore {
      */
     void addFiles(List<FileReference> fileReferences) throws StateStoreException;
 
+    /**
+     * Adds files to the Sleeper table, with any number of references.
+     * <p>
+     * Each new file should be specified once, with all its references.
+     * <p>
+     * A file must never be referenced on two partitions where one is a descendent of another. This means each record in
```

Comment:
I think ` A file must never be referenced in two partitions` or ` A file must never be referenced from two partitions` would sound better.

Reply:
I've made that change, thanks